### PR TITLE
feat: forum topic lock/unlock

### PR DESCRIPTION
**Pull request by DaStormer**
Please describe the changes this PR makes and why it should be merged:

Added support for locking/unlocking forum topics

Status

-   [ ] Code changes have been tested.

Semantic versioning classification:

-   [x] This PR changes the library's interface (methods or parameters added)
-   [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
-   [ ] This PR only includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts [hunk 1]
```diff
@@ -26,28 +22,38 @@ export class ForumEventHandler extends GatewayEventHandler {
...
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_PINNED, getCachedtopic ?? new ForumTopic(this.client, data.d.forumTopic));
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
+        getCachedTopic?._update({ isPinned: true });
+        return this.client.emit(constants.clientEvents.FORUM_TOPIC_PINNED, getCachedTopic ?? new ForumTopic(this.client, data.d.forumTopic));
     }
     forumTopicUnpinned(data: WSForumTopicUnpinned) {
-        const getCachedtopic = this.client.topics.cache.get(data.d.forumTopic.id);
-        getCachedtopic?._update({ isPinned: false });
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_UNPINNED, getCachedtopic ?? new ForumTopic(this.client, data.d.forumTopic));
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
```
Comment (zaida04) [inline packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts:45]: Should we abstract this kinda logic into a helper function or method so we don't have to duplicate this line in every event? Or maybe we should have a step before this event function is called where we check the cache and pass it through as a parameter in this method.

Comment (DaStormer) [reply packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts:45]: Not sure how a helper function would be useful for this. You'd still need to call that func on every event right

Comment (zaida04) [reply packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts:45]: Let me rephrase, what if we had a step BEFORE calling methods like forumTopicUpdated where we get the cached structure and pass it as a parameter into forumTopicUpdated.

Comment (DaStormer) [reply packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts:45]: Ah. I feel like in this case the abstraction is unnecessary and maybe even counterproductive since it would create an avoidable inconsistency with how the forum events are handled vs all others (there would be an additional step).

If you think this would be better though, sure I don't mind.

**Review by zaida04 — APPROVED** [review]
